zerver/tests/test_bugdown.py

```diff
@@ -1638,6 +1638,41 @@ def test_md_relative_link(self) -> None:
             '<p><a href="#narrow/stream/999-hello" title="#narrow/stream/999-hello">hello</a></p>'
         )
 
+    def test_github_link(self) -> None:
+        realm = get_realm("zulip")
+        sender_user_profile = self.example_user('othello')
+        message = Message(sender=sender_user_profile, sending_client=get_client("test"))
+
+        msg = "https://github.com/zulip/zulip"
+        self.assertEqual(
+            bugdown.convert(msg, message_realm=realm, message=message),
+            '<p><a href="https://github.com/zulip/zulip" target="_blank" title="https://github.com/zulip/zulip">zulip/zulip</a></p>'
+        )
+
+        msg = "https://github.com/zulip/zulip/issues/11895"
+        self.assertEqual(
+            bugdown.convert(msg, message_realm=realm, message=message),
+            '<p><a href="https://github.com/zulip/zulip/issues/11895" target="_blank" title="https://github.com/zulip/zulip/issues/11895">zulip/zulip#11895</a></p>'
+        )
+
+        msg = "https://github.com/zulip/zulip/pull/11922"
+        self.assertEqual(
+            bugdown.convert(msg, message_realm=realm, message=message),
+            '<p><a href="https://github.com/zulip/zulip/pull/11922" target="_blank" title="https://github.com/zulip/zulip/pull/11922">zulip/zulip#11922</a></p>'
+        )
+
+        msg = "https://github.com/zulip/zulip/commit/a47edd4fc8a1b115e979bef6fc372ddf360cdcef"
+        self.assertEqual(
+            bugdown.convert(msg, message_realm=realm, message=message),
+            '<p><a href="https://github.com/zulip/zulip/commit/a47edd4fc8a1b115e979bef6fc372ddf360cdcef" target="_blank" title="https://github.com/zulip/zulip/commit/a47edd4fc8a1b115e979bef6fc372ddf360cdcef">zulip/zulip@a47edd4</a></p>'
+        )
+
+        msg = "https://github.com/zulip/zulip/pull/11922/commits/623ee15beef4bd8745e9f4746fcc11093909dcc3"
+        self.assertEqual(
+            bugdown.convert(msg, message_realm=realm, message=message),
+            '<p><a href="https://github.com/zulip/zulip/pull/11922/commits/623ee15beef4bd8745e9f4746fcc11093909dcc3" target="_blank" title="https://github.com/zulip/zulip/pull/11922/commits/623ee15beef4bd8745e9f4746fcc11093909dcc3">zulip/zulip@623ee15</a></p>'
+        )
```

Comment:
I'm not sure the syntax for this last one is a good idea.  I realize it's a GitHub feature, but I feel like it sorta creates the expectation that the commit is actually merged.  It might make sense to just not auto-shorten things of this form:

`https://github.com/zulip/zulip/pull/11922/commits/623ee15beef4bd8745e9f4746fcc11093909dcc3`

or at least have the text be something more like:

`zulip/zulip/#11922:623ee15bee`

that makes clear it's part of a PR.

Reply:
That's a fair point. For now it was easiest to just not shorten the links. I'd also guess that pasting these links will be quite rare so they won't clutter messages too much.